backend/convert_to_protobuf.go

```diff
@@ -141,6 +141,10 @@ func (t ConvertToProtobuf) QueryDataResponse(res *QueryDataResponse) (*pluginv2.
 	}
 	for refID, dr := range res.Responses {
 		for _, f := range dr.Frames {
+			// Don't process nil frames
```

Comment:
Do we risk hiding errors by doing this? If developers shouldn't add nil frames, wouldn't it be better to fail fast?

Reply:
Then we should log an error, but failing by trying to access a field that's nil? That's not good either. We should have an explicit message in the logs then that say "Somebody did something bad"

Reply:
Actually, I realized this is probably server-side(?). Meaning, the opportunity to fail fast has kind of passed. In that case, logging the error would be for the plugin user rather than for the plugin author? Don't have a strong opinion on this, I'll defer to @kylebrandt here.

Reply:
I've updated the code to return an error. Thoughts?

Reply:
What happens on the frontend of a some dataframes are returned, and one of the entries in the slice is `null` (or at least, this how I would expect it encode?

There shouldn't be a nil frame in the response. But that would be on the plugin developer to handle (put a frame with a notice of error/warning severity attached to it - I think). But this will happen at runtime and devs might likely not catch it, so probably better not to push the error to the user.

Logging a warning (or maybe an error - depending on what the frontend does seems reasonable). Then at least bug reports can get back to the devs if there are annoying log entries, but won't prevent not working at all (assuming the frontend will show the other frames if one is null).

In the frontend will not handle it a null frame. Then I think we should log a warning, and change the above code to append to a slice instead of preallocating so there are not any nulls in the slice of encoded frames.